### PR TITLE
Re-enable solver tests and update petsc to 3.13.1

### PR DESCRIPTION
**Pull request by cekees**
# Mandatory Checklist

Please ensure that the following criteria are met:

- [x] Title of pull request describes the changes/features
- [x] Request at least 2 reviewers
- [x] If new files are being added, the files are no larger than 100kB. Post the file sizes.
- [x] Code coverage did not decrease. If this is a bug fix, a test should cover that bug fix. If a new feature is added, a test should be made to cover that feature.
- [x] New features have appropriate documentation strings (readable by sphinx)
- [x] Contributor has read and agreed with CONTRIBUTING.md and has added themselves to CONTRIBUTORS.md 

As a general rule of thumb, try to follow [PEP8](https://www.python.org/dev/peps/pep-0008/) guidelines.

# Description

### CODE REVIEW

**Comment by zhang-alvin** [comment]
@cekees, there seems to be a file conflict with `test_quads.py`?

File: proteus/mprans/SW2DCV.py [hunk 1]
```diff
@@ -356,7 +356,7 @@ def evaluate(self, t, c):
     def preStep(self, t, firstStep=False):
         if firstStep:
             # Init boundaryIndex
-            assert (self.model.boundaryIndex is None and self.model.normalx is not None, "Check boundaryIndex, normalx and normaly")
+            assert self.model.boundaryIndex is None and self.model.normalx is not None, "Check boundaryIndex, normalx and normaly"
```
Comment (zhang-alvin) [inline proteus/mprans/SW2DCV.py]: didn't @ejtovar observe the opposite behavior being necessary for python3?

File: proteus/tests/periodic/test_periodic.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ def test_periodic_2D(load_periodic_opts_2D,
     expected_path = 'comparison_files/' + 'comparison_' + 'ductq1t12dpghe0.0975609756097561' + '_velocity_t25.csv'
     #write comparison file
     #np.array(actual.root.velocity_t25).tofile(os.path.join(script_dir, expected_path),sep=",")
-    np.testing.assert_almost_equal(np.fromfile(os.path.join(script_dir, expected_path),sep=","),np.array(actual.root.velocity_t25).flatten(),decimal=8)
+    np.testing.assert_almost_equal(np.fromfile(os.path.join(script_dir, expected_path),sep=","),np.array(actual.root.velocity_t25).flatten(),decimal=7)
```
Comment (zhang-alvin) [inline proteus/tests/periodic/test_periodic.py:93]: Is it just petsc's internal methods that are driving the slight changes in the results?

**Comment by cekees** [comment]
I see, that came from the os x PR merge a few minutes ago. I fixed it. I imagine we may have some os x test failures, but I'm just going to disable/skip any that arise for now.

File: proteus/tests/solver_tests/test_nseDC_THQuad_2D.py [hunk 3]
```diff
@@ -63,13 +55,12 @@ def _runTest(self):
...
         # be updated accordingly.
         relpath = 'comparison_files/drivenCavityNSE_LSC_expected.h5'
-        expected = tables.open_file(os.path.join(self._scriptdir,relpath))
         actual = tables.open_file('drivenCavityNSETrial.h5','r')
-
-        assert numpy.allclose(expected.root.velocaity_t7.read(),
-                              actual.root.velocity_t7.read(),
-                              atol=1e-2) 
-        expected.close()
+        np.savetxt(os.path.join(self._scriptdir,relpath),actual.root.velocity_t7.read(),delimiter=',')
```
Comment (zhang-alvin) [inline proteus/tests/solver_tests/test_nseDC_THQuad_2D.py:59]: What's the motivation for writing the text file here? It doesn't look like the current test really requires it as the data is in memory already.

File: proteus/mprans/SW2DCV.py (hunk 1 above)
Comment (cekees) [reply proteus/mprans/SW2DCV.py]: I thought he saw the same. The conda build  of python seems to do a better job of explaining why, but basically assert is a keywork, not a function, and no-empty tuples always evaluate to True, so this is a nefarious bug. If you want to do multi-line use \ or be careful to enclose only the test in (): 

https://stackoverflow.com/questions/3112171/python-assert-with-and-without-parenthesis

File: proteus/tests/solver_tests/test_nseDC_THQuad_2D.py (hunk 3 above)
Comment (cekees) [reply proteus/tests/solver_tests/test_nseDC_THQuad_2D.py:59]: I meant to fix that. Thanks for catching it. Will push one more commit shortly.

File: proteus/mprans/SW2DCV.py (hunk 1 above)
Comment (zhang-alvin) [reply proteus/mprans/SW2DCV.py]: Okay I see. I've seen a lot of `raise(NameError, 'string')` statements in setup files so it looked like the parentheses were supposed to be there.

**Comment by cekees** [comment]
> Looks good. Again, just a reminder on file size.

Roger on the file size. I had been aiming at < 200kb before I released our target was actually 100kb. Anyway, I mostly got rid of comparison files here and switched to csv. I merged all these solver_tests into one place for a future effort do a better job on test criteria and consistency of options, use of fixtures, etc.

**Pull request merged**